db/migrate/20220723075636_add_user_ref_to_post.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :author, null: false, foreign_key: true { to_table: :users }
+  end
```

Comment:
Kindly, make sure migration works correctly I'm getting error while I migrate the tables. You can see the error below.

<img width="1292" alt="image" src="https://user-images.githubusercontent.com/34578631/180599715-4f6ff97f-cf75-4fdc-81a3-f8288c829e1b.png">

Try removing `true` after `foreign_key` hope that will fix the issues and I notice you have the same thing in `like` migration too.